backend/functions/lib/adapters/gateways/user-session-cache.cosmosdb.repository.ts

```diff
@@ -0,0 +1,62 @@
+import { ApplicationContext } from '../types/basic';
+import { CosmosDbRepository } from './cosmos/cosmos.repository';
+import { CamsSession } from '../../../../../common/src/cams/session';
+import { UserSessionCacheRepository } from './user-session-cache.repository';
+import { JwtClaims } from '../types/authorization';
+
+const MODULE_NAME: string = 'COSMOS_DB_REPOSITORY_USER_SESSION_CACHE';
+const CONTAINER_NAME: string = 'user-session-cache';
+
+type CachedCamsSession = CamsSession & {
+  id?: string;
+  signature: string;
+  ttl: number;
+};
+
+export class UserSessionCacheCosmosDbRepository implements UserSessionCacheRepository {
+  private repo: CosmosDbRepository<CachedCamsSession>;
+  constructor(context: ApplicationContext) {
+    this.repo = new CosmosDbRepository<CachedCamsSession>(context, CONTAINER_NAME, MODULE_NAME);
+  }
+
+  public async get(context: ApplicationContext, token: string): Promise<CamsSession | null> {
+    const signature = token.split('.')[2];
+    const query = 'SELECT * FROM c WHERE c.signature = @signature';
+    const querySpec = {
+      query,
+      parameters: [
+        {
+          name: '@signature',
+          value: signature,
+        },
+      ],
+    };
+
+    const cached = await this.repo.query(context, querySpec);
+    if (cached && cached.length === 1) {
+      return toCamsSession(cached[0]);
+    } else {
+      return null;
+    }
+  }
+
+  public async put(context: ApplicationContext, session: CamsSession): Promise<CamsSession> {
+    // TODO: handle input that cannot possibly be a jwt?
```

Comment:
??

Reply:
will there ever be a case like this?

Reply:
Similar to the changes we recently made to the authentication configuration, if the input to this happened to contain a "token" that was not a jwt, we could get unexpected behavior as we split up the string expecting two `.`'s to separate the sections.